### PR TITLE
Fix formatting that has got out of sync and set things up so this won't happen again

### PR DESCRIPTION
**Pull request by DRMacIver**
Because of the way we cache the tools virtualenv, make format can behave differently locally and on travis. Obviously, this is bad.

This PR does several things:

* Fixes the formatting so that we're up to date with the latest version of pyformat etc
* Pins the versions of the tools we depend on so that we depend on a fixed version
* Configure pyup.io to give us pull requests when new versions of those dependencies come out

#465 should be merged first (I'm just waiting on a green build) and then I'll update this pull request to also configure test dependencies to be updated. Afterwards I'll do a follow-up pull request for the rest of our testing dependencies.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -63,11 +62,13 @@ $(PY36):
 $(PYPY):
 	scripts/retry.sh scripts/install.sh pypy
 
-$(TOOL_VIRTUALENV): $(PY34)
+$(TOOL_VIRTUALENV): $(PY34) requirements/tools.txt
+	rm -rf $(TOOL_VIRTUALENV)
```
Comment (alexwlchan) [inline Makefile]: The `documentation` make task installs sphinx and sphinx-rtd-theme in this virtualenv, then calls this Make task. I think you’re blowing away Sphinx after it’s installed. Needs fixing before we can get a clean run of the tests.

Comment (DRMacIver) [inline Makefile]: Good catch, thanks. Fixed.

**Comment by DRMacIver** [comment]
As well as fixing it so that we don't nuke sphinx I've also:

* Rationalized isort into the same virtualenv (I'm sure past-DRMacIver had a good reason for separating them, but if so he neglected to tell me what it was)
* Stopped installing Hypothesis into the tool virtualenv (!) and just set PYTHONPATH appropriately when building docs.

**Comment by DRMacIver** [comment]
Hmm. Which apparently broke a tonne of stuff that worked when I tried it locally. Let's try that again...

**Comment by alexwlchan** [comment]
🎉

**Pull request merged**